### ISSUE TITLE
"Click here to view full version" leads to blank page if not logged in to Skype in web browser

### ISSUE THREAD

**Issue by dreamlayers** [EionRobb/skype4pidgin#498]
An image I sent gave me two "Click here to view full version" links. The first link was a proper https://eus1-api.asm.skype.com/v1/objects/<something>/views/imgpsh_fullsize link and the second just had the hostname and did nothing when clicked. That first link led me to a blank page. So I opened the old Linux Skype client, and clicked on the link there. It asked me to log in, and after that showed me the image. Then I could click on the first link in Pidgin and see the image. This is in skypeweb 1.2.2.

This could be related to #386, because I also did not see a thumbnail.

**Comment by alexolog** [comment]
Confirmed.  Although I did see the thumbnail

**Comment by bolera** [comment]
I'm not seeing anything. I just get a link that opens a blank page in the browser. Wouldn't it be possible to just pull in the image? I mean I'm logged in, am I?  (btw, might be helpful to add the version number to the dll. I just checked Properties/Details and there is no version information.

**Comment by EionRobb** [comment]
You won't be able to click through to the image if you haven't signed into web.skype.com in the past 24 hours as you won't have the correct cookies in your browser to have auth to be able to view.  Once you login there you should be able to refresh your browser to see the images.  I haven't yet found a workaround other than a way that forces you to login for each image (even if you're already logged in)

However, the thumbnails are working more reliably for me since the new HTTP code 1.4 - @bolera can you check in Help->Plugin Information (from the buddy list) to see what version number "Skype (HTTP)" is?  It can be possible to have multiple copies of the dll installed, but only the oldest one being loaded.

**Comment by bolera** [comment]
Yes, it can, thanks for the hint. But it took me a while to find out *what* I had done incorrectly as not even using your installer made the 1.4 appear. I had renamed the old skypeweb.dll, but left it in the plugins directory. I wasn't aware or actually had forgotten that Pidgin would try to load everything there. So, I kept seeing only 1.2.2 in the plugin information. Once I removed the file that changed. I now got one thumbnail, but not at the right position but as the latest addition in a chat room. I assume that's because of the libpurple limitation to change inline HTML, that is mentioned in another report.
I've also noticed that the duplicating of the protocol seems to have stopped.
I've also found out about a caveat with the login. Actually, the browser would sign me automatically in, but I have two Skype accounts. One "skypename" account and one "live:" account. For Skype I usually use the skypename account, but for all other things MSA I use the live-Account. So, that is basically logged in all the time. So, when I visit web.skype.com it automatically logs me in with that live account. And if I stop this and sign in with the skypename account it wants to log me in to other MS services with that account - which doesn't exist as an MSA. I guess I have to merge the two. I didn't do until now because I had some guts feeling I shouldn't do that.

**Comment by kotarou3** [comment]
It would be nice if the client just showed the full size image directly in the chatbox instead (or have an option to do so), so I could quickly view the full image just by maximising the chat window

**Comment by Penaz91** [comment]
After a couple of tests I think it's possible to force the browser to log in by building a specific URL.
Given then internal URL:
https://eus1-api.asm.skype.com/v1/objects/<id>/views/imgpsh_fullsize

We can build a skypeweb xmmfallback link:
https://web.skype.com/xmmFallback?docid=<id>

Which triggers the login automatically.

Alternatively we can use:
https://login.skype.com/login/sso?go=webclient.xmm&docid=<id>

Hopefully this helps a bit.

**Comment by oprudkyi** [comment]
@Penaz91 
+1 for https://web.skype.com/xmmFallback?docid=[id] 
(this is how skype 4.3 show links, so it is more or less "official")

**Comment by Penaz91** [comment]
Skype 4.3 is deprecated now, so I have no idea how much this way of seeing things will "stay around", since now the company that owns skype made the web-based version canon for Linux.

**Comment by oprudkyi** [comment]
as for today it still works (that was the only way till I discovered the plugin for me to have text chats in separate windows)

**Comment by oprudkyi** [comment]
well, disabling "Conversation Colors" plugin (or disabling "Ignore input format" in it) will show images embed in the chat window
suggested by https://github.com/majn/telegram-purple#faq
